client/src/components/Login.js

```diff
@@ -0,0 +1,311 @@
+
+    if (method) {
+      this.setSelectedMethod(
+        registeredMethods.find(methodSpec => methodSpec.urlSegment === method)
+      );
+    }
+  }
+
+  /**
+   * Render a control that will allow a user to display the "other methods" pane if the currently
+   * selected method is not suitable
+   *
+   * @return {HTMLElement|null}
+   */
+  renderOtherMethodsControl() {
+    const otherMethods = this.getOtherMethods();
+    const { ss: { i18n } } = window;
+
+    // There shouldn't be a control if there are no other methods to choose from
+    if (!Array.isArray(otherMethods) || !otherMethods.length) {
+      return null;
+    }
+
+    return (
+      <a href="#" onClick={this.handleShowOtherMethodsPane}>
+        {i18n._t('MFALogin.MORE_OPTIONS', 'More options')}
+      </a>
+    );
+  }
+
+  /**
+   * If the half-logged in member has more than one authentication method set up, show a list of
+   * others they have enabled that could also be used to complete authentication and log in.
+   *
+   * @return {HTMLElement|null}
+   */
+  renderOtherMethods() {
+    const otherMethods = this.getOtherMethods();
+    const { ss: { i18n } } = window;
+
+    return (
+      <div className="mfa-login__other-methods">
+        <h2>{i18n._t('MFALogin.OTHER_METHODS_TITLE', 'Try another way to verify')}</h2>
+        <ul>
+          {otherMethods.map(method => (
+            <li key={method.urlSegment}>
+              <button onClick={this.handleClickOtherMethod} value={method.urlSegment}>
+                {method.name}
+              </button>
+            </li>
+          ))}
+        </ul>
+        <p>
+          {i18n._t(
+            'MFALogin.LAST_RESORT_MESSAGE',
+            'Contact your site administrator if you require your multi-factor authentication to ' +
+              'be reset'
+          )}
+        </p>
+        <button
+          className="mfa-login__other-methods-back"
+          onClick={this.handleHideOtherMethodsPane}
+        >
+          Back
+        </button>
+      </div>
+    );
+  }
+
+  /**
+   * Render the component for the currently selected method
+   *
+   * @return {HTMLElement}
+   */
+  renderSelectedMethod() {
+    const { selectedMethod, loginProps, message } = this.state;
+
+    const MethodComponent = loadComponent(selectedMethod.component);
+
+    return (
+      <div>
+        <h2>{selectedMethod.leadInLabel}</h2>
+        {MethodComponent && <MethodComponent
+          {...loginProps}
+          method={selectedMethod}
+          error={message}
+          onCompleteLogin={this.handleCompleteLogin}
+          moreOptionsControl={this.renderOtherMethodsControl()}
+        />}
+      </div>
+    );
+  }
+
+  render() {
+    const { loading, selectedMethod, showOtherMethods } = this.state;
+
+    if (loading) {
+      return <div className="mfa__loader" />;
+    }
+
+    if (showOtherMethods || !selectedMethod) {
+      return this.renderOtherMethods();
+    }
+
+    return this.renderSelectedMethod();
```

Comment:
I think I'd rather see this logic flipped. "If you've selected a method then render it, otherwise show all"